### PR TITLE
doc/user: attempt to clarify source stats semantics

### PR DESCRIPTION
**Pull request by benesch**
Per @teskje's feedback.

### Motivation

   * This PR refactors existing docs.

### Checklist

- [x] This PR has adequate test coverage / QA involvement has been duly considered. ([trigger-ci for additional test/nightly runs](https://trigger-ci.dev.materialize.com/))
- [x] This PR has an associated up-to-date [design doc](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/README.md), is a design doc ([template](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/00000000_template.md)), or is sufficiently small to not require a design.
  
- [x] If this PR evolves [an existing `$T ⇔ Proto$T` mapping](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/command-and-response-binary-encoding.md) (possibly in a backwards-incompatible way), then it is tagged with a `T-proto` label.
- [x] If this PR will require changes to cloud orchestration or tests, there is a companion cloud PR to account for those changes that is tagged with the release-blocker label ([example](https://github.com/MaterializeInc/cloud/pull/5021)).
  
- [x] If this PR includes major [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note), I have pinged the relevant PM to schedule a changelog post.

### CODE REVIEW

File: doc/user/content/sql/system-catalog/mz_internal.md [hunk 1]
```diff
@@ -1045,80 +1045,82 @@ messages and additional metadata helpful for debugging.
...
+| `bytes_received`          | [`uint8`]    | The number of bytes the source has read from the external system. Bytes are counted in a source type-specific manner and may or may not include protocol overhead. |
+| `updates_staged`          | [`uint8`]    | The number of updates (insertions plus deletions) the source has written but not yet committed to the storage layer. |
+| `updates_committed`       | [`uint8`]    | The number of updates (insertions plus deletions) the source has committed to the storage layer. |
+| `records_indexed`         | [`uint8`]    | The number of individual records indexed in the source envelope state. |
+| `bytes_indexed`           | [`uint8`]    | The number of bytes stored in the source's internal index, if any. |
+| `rehydration_latency`     | [`interval`] | The amount of time it took for the source to rehydrate its internal index, if any, after the source last restarted. |
+| `snapshot_records_known`  | [`uint8`]    | The size of the source's snapshot, measured in number of records. See below to learn what constitutes a record. |
```
Comment (kay-kim) [inline doc/user/content/sql/system-catalog/mz_internal.md:1058]: Nit (feel free to ignore): To be in line with the next row, maybe:

The total number of records in the source's snapshot (i.e., the size of the snapshot in records). or some such?

File: doc/user/content/sql/system-catalog/mz_internal.md [hunk 2]
```diff
@@ -1045,80 +1045,82 @@ messages and additional metadata helpful for debugging.
...
 
-`records_indexed` and `bytes_indexed` are the size (in records and bytes respectively) of the data the given source _indexes_. Currently, this is only
-`UPSERT` and `DEBEZIUM` sources. These reset to 0 when sources are restarted and must re-index their data.
+### Gauge metrics
 
----
-**Rehydration latency**
+Gauge metrics reflect values that can increase or decrease over time. Gauge
+metrics are eventually consistent. They may lag the true state of the source by
+seconds or minutes, but if the source stops ingesting messages the gauges will
```
Comment (kay-kim) [inline doc/user/content/sql/system-catalog/mz_internal.md:1085]: nit (feel free to ignore)

```suggestion
seconds or minutes, but if the source stops ingesting messages, the gauges will
```

File: doc/user/content/sql/system-catalog/mz_internal.md [hunk 3]
```diff
@@ -1045,80 +1045,82 @@ messages and additional metadata helpful for debugging.
...
+is the offset that Materialize has durably ingested. These metrics will never
+decrease over the lifetime of a source.
 
-These statistics have a source-defined unit:
-- For Kafka sources, its the number of offsets.
-- For MySQL sources, its the number of transactions.
-- For Postgres sources, its the number of bytes in its replication stream.
+The meaning of offset depends on the source:
+- For Kafka sources, an offset is the Kafka message offset.
+- For MySQL sources, an offset is the number of transactions committed across all servers in the cluster.
```
Comment (kay-kim) [inline doc/user/content/sql/system-catalog/mz_internal.md:1122]: Q: Should we append with  `(GTID)` --  (https://materializeinc.slack.com/archives/CQM08AP9V/p1733296713840719?thread_ts=1733258115.549209&cid=CQM08AP9V ?

**Review by kay-kim — APPROVED** [review]
lgtm mod feel-free-to-ignore nits.

File: doc/user/content/sql/system-catalog/mz_internal.md (hunk 3 above)
Comment (benesch) [reply doc/user/content/sql/system-catalog/mz_internal.md:1126]: It's not actually a GTID. It's weirder than that. A GTID is a combo of a server UUID and a transaction number. Multiple servers can be committing transactions at the same time. So we sum up the transaction numbers across all servers. In the general case, it doesn't really tie back to anything in MySQL land unfortunately. (Although if you're only using one server it'll be the transaction number of that one server.)

Comment (kay-kim) [reply doc/user/content/sql/system-catalog/mz_internal.md:1126]: 👍 Thanks for the clarification!

**Pull request merged**